uspace/lib/ddev/src/ddev_srv.c

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2019 Jiri Svoboda
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * - Redistributions of source code must retain the above copyright
+ *   notice, this list of conditions and the following disclaimer.
+ * - Redistributions in binary form must reproduce the above copyright
+ *   notice, this list of conditions and the following disclaimer in the
+ *   documentation and/or other materials provided with the distribution.
+ * - The name of the author may not be used to endorse or promote products
+ *   derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/** @addtogroup libddev
+ * @{
+ */
+/**
+ * @file
+ * @brief Display protocol server stub
+ */
+
+#include <ddev_srv.h>
+#include <errno.h>
+#include <io/log.h>
+#include <ipc/ddev.h>
+#include <mem.h>
+#include <stdlib.h>
+#include <stddef.h>
+
+#include <stdio.h>
+
+/** Connect to a GC.
+ *
+ * XXX As a workaround here we tell the client the values of arg2 and arg3
+ * needed to connect to the GC using async_connect_me_to(), these need
+ * to be provided by the ddev_ops_t.get_gc. Different values are needed
+ * for a DDF driver or a regular server. This would not be needed if we
+ * had a proper way of creating an endpoint and passing it to our client.
+ */
+static void ddev_get_gc_srv(ddev_srv_t *srv, ipc_call_t *icall)
+{
+	sysarg_t arg2;
+	sysarg_t arg3;
+	errno_t rc;
+
+	printf("ddev_get_gc_srv\n");
+
+	if (srv->ops->get_gc == NULL) {
+		async_answer_0(icall, ENOTSUP);
+		return;
+	}
+
+	rc = srv->ops->get_gc(srv->arg, &arg2, &arg3);
+	async_answer_2(icall, rc, arg2, arg3);
+}
+
+/** Get display device information */
+static void ddev_get_info_srv(ddev_srv_t *srv, ipc_call_t *icall)
+{
+	ddev_info_t info;
+	errno_t rc;
+
+	printf("ddev_get_info_srv\n");
```

Comment:
debug print

Reply:
Hm, there are more printfs in this file. Is this by intention or is this a left-over?

Reply:
Fixed.